File: pydss/delivery/WindingImpl.py
from dss.delivery.Winding import (Winding,)
from dss.common.impl.DSSGlobals import (DSSGlobals,)


class WindingImpl(object, Winding):
    connection = None
    kVLL = None
    VBase = None
    kVA = None
    puTap = None
    Rpu = None
    RNeut = None
    XNeut = None
    # on transformer MVABase (1st winding)
    Y_PPM = None
    # Anti float reactance adder
    # Tap changer data
    tapIncrement = None
    minTap = None
    maxTap = None
    numTaps = None

    def __init__(self):
        super(WindingImpl, self)()
        self.connection = 0
        self.kVLL = 12.47
        self.VBase = (self.kVLL / DSSGlobals.SQRT3) * 1000.0
        self.kVA = 1000.0
        self.puTap = 1.0
        self.Rpu = 0.002
        self.RNeut = -1.0
        # default to open - make user specify connection
        self.XNeut = 0.0
        self.computeAntiFloatAdder(1e-06, self.kVA / 3.0 / 1000.0)
        # 1 PPM
        self.tapIncrement = 0.00625
        self.numTaps = 32
        self.maxTap = 1.1
        self.minTap = 0.9

    def computeAntiFloatAdder(self, PPM_Factor, VABase1ph):
        self.Y_PPM = -PPM_Factor / self.Math.pow(self.VBase, 2) / VABase1ph

    def getConnection(self):
        return self.connection

    def setConnection(self, conn):
        self.connection = conn

    def getKVLL(self):
        return self.kVLL

    def setKVLL(self, kvll):
        self.kVLL = kvll

    def getVBase(self):
        return self.VBase

    def setVBase(self, base):
        self.VBase = base

    def getKVA(self):
        return self.kVA

    def setKVA(self, kva):
        self.kVA = kva

    def getPUTap(self):
        return self.puTap

    def setPUTap(self, tap):
        self.puTap = tap

    def getRpu(self):
        return self.Rpu

    def setRpu(self, rpu):
        self.Rpu = rpu

    def getRNeut(self):
        return self.RNeut

    def setRNeut(self, rneut):
        self.RNeut = rneut

    def getXNeut(self):
        return self.XNeut

    def setXNeut(self, xneut):
        self.XNeut = xneut

    def getY_PPM(self):
        return self.Y_PPM

    def setY_PPM(self, y):
        self.Y_PPM = y

    def getTapIncrement(self):
        return self.tapIncrement

    def setTapIncrement(self, increment):
        self.tapIncrement = increment

    def getMinTap(self):
        return self.minTap

    def setMinTap(self, min):
        self.minTap = min

    def getMaxTap(self):
        return self.maxTap

    def setMaxTap(self, max):
        self.maxTap = max

    def getNumTaps(self):
        return self.numTaps

    def setNumTaps(self, num):
        self.numTaps = num

File: core/dss-core/src/main/java/com/ncond/dss/delivery/impl/WindingImpl.java
package com.ncond.dss.delivery.impl;

import com.ncond.dss.common.impl.DSS;
import com.ncond.dss.delivery.Winding;

public class WindingImpl implements Winding {

	private int connection;
	private double kVLL,
		VBase,
		kVA,
		puTap,
		Rpu,  // on transformer MVABase (1st winding)
		RNeut,
		XNeut;
	private double Y_PPM;  // Anti float reactance adder

	/* Tap changer data */
	private double tapIncrement,
		minTap,
		maxTap;
	private int numTaps;

	public WindingImpl() {
		super();
		connection = 0;
		kVLL       = 12.47;
		VBase      = kVLL / DSS.SQRT3 * 1000.0;
		kVA        = 1000.0;
		puTap      = 1.0;
		Rpu        = 0.002;
		RNeut      = -1.0;  // default to open - make user specify connection
		XNeut      = 0.0;
		computeAntiFloatAdder(1.0e-6, kVA / 3.0 / 1000.0);  //  1 PPM

		tapIncrement = 0.00625;
		numTaps      = 32;
		maxTap       = 1.10;
		minTap       = 0.90;
	}

	@Override
	public void computeAntiFloatAdder(double PPM_Factor, double VABase1ph) {
		Y_PPM = -PPM_Factor / (Math.pow(VBase, 2) / VABase1ph);
	}

	@Override
	public int getConnection() {
		return connection;
	}

	@Override
	public void setConnection(int conn) {
		connection = conn;
	}

	@Override
	public double getKVLL() {
		return kVLL;
	}

	@Override
	public void setKVLL(double kvll) {
		kVLL = kvll;
	}

	@Override
	public double getVBase() {
		return VBase;
	}

	@Override
	public void setVBase(double base) {
		VBase = base;
	}

	@Override
	public double getKVA() {
		return kVA;
	}

	@Override
	public void setKVA(double kva) {
		kVA = kva;
	}

	@Override
	public double getPUTap() {
		return puTap;
	}

	@Override
	public void setPUTap(double tap) {
		puTap = tap;
	}

	@Override
	public double getRpu() {
		return Rpu;
	}

	@Override
	public void setRpu(double rpu) {
		Rpu = rpu;
	}

	@Override
	public double getRNeut() {
		return RNeut;
	}

	@Override
	public void setRNeut(double rneut) {
		RNeut = rneut;
	}

	@Override
	public double getXNeut() {
		return XNeut;
	}

	@Override
	public void setXNeut(double xneut) {
		XNeut = xneut;
	}

	@Override
	public double getY_PPM() {
		return Y_PPM;
	}

	@Override
	public void setY_PPM(double y) {
		Y_PPM = y;
	}

	@Override
	public double getTapIncrement() {
		return tapIncrement;
	}

	@Override
	public void setTapIncrement(double increment) {
		tapIncrement = increment;
	}

	@Override
	public double getMinTap() {
		return minTap;
	}

	@Override
	public void setMinTap(double min) {
		minTap = min;
	}

	@Override
	public double getMaxTap() {
		return maxTap;
	}

	@Override
	public void setMaxTap(double max) {
		maxTap = max;
	}

	@Override
	public int getNumTaps() {
		return numTaps;
	}

	@Override
	public void setNumTaps(int num) {
		numTaps = num;
	}

}
